packages/mongodb-memory-server-core/src/util/__tests__/MongoInstance.test.ts

```diff
@@ -292,12 +292,31 @@ describe('MongodbInstance', () => {
     await mongod.start();
     jest.spyOn(MongoClient, 'connect');
     process.kill(mongod.mongodProcess!.pid, 'SIGKILL');
-    await new Promise<void>((resolve) => setTimeout(resolve, 100));
+    while (dbUtil.isAlive(mongod.mongodProcess!.pid)) {
+      await new Promise<void>((resolve) => setTimeout(resolve, 5));
+    }
     await mongod.stop();
 
     expect(MongoClient.connect).not.toBeCalled();
   });
 
+  test('"kill" should not wait too much to open a connection', async () => {
+    const gotPort = await getFreePort();
+    const mongod = new MongodbInstance({
+      instance: {
+        replSet: 'testset',
+        ip: '127.0.0.1',
+        port: gotPort,
+        dbPath: tmpDir,
+      },
+    });
+    await mongod.start();
+    jest.spyOn(MongoClient, 'connect');
+    process.kill(mongod.mongodProcess!.pid, 'SIGKILL');
+    await mongod.stop();
+    expect(MongoClient.connect).toBeCalled();
+  }, 8000); // the default serverSelectionTimeoutMS is 10s, so waiting for 8s seems ok to catch it
```

Comment:
> the default serverSelectionTimeoutMS is 10s, so waiting for 8s seems ok to catch it

isnt the default 30 seconds?
https://github.com/mongodb/node-mongodb-native/blob/2ab2189561736e354437df88d2ea26cf35b483d1/src/connection_string.ts#L1078-L1081

Reply:
```suggestion
  }, 8000); // the default serverSelectionTimeoutMS is 30s, so waiting for 8s seems ok to catch it
```